data-generator.tpl.ts

```diff
@@ -84,13 +92,59 @@ const createFoodProductImageValue = {
   },
 };
 
+const rawImageUrlToDownloadUrl = (imageUrl: string) => {
+  const width = 200;
+  const height = width * 2;
+  return `${imageUrl}?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=${width}&h=${height}&q=80&fm=jpg`;
+};
+
 const createBookImageUrlValue = {
   createValue: (seed: number, _refTime: number) => {
-    const width = 200;
-    const height = width * 2;
     const imageBackgroundUrl = random(options['BookImageBackground.src'], seed);
+    return rawImageUrlToDownloadUrl(imageBackgroundUrl);
+  },
+};
+
+const waitForImageFetchingToFinish = () => {
+  return new Promise<void>((resolve) => {
+    (function waitForFlagChange() {
+      if (!imageFetchingInProgress) return resolve();
+      setTimeout(waitForFlagChange, 50);
+    })();
+  });
```

Comment:
Please don't ask me to explain this code. I copied it from SO after struggling for a day to force the different async calls to wait for the first one to finish fetching the data.